DOCSP-22844: POJO CRUD examples (#512)
mongodb/docs-java: source/fundamentals/data-formats/document-data-format-pojo.txt

Configure the Driver for POJOs
------------------------------

To configure the driver to use POJOs, you must specify the following
components:

> - ``PojoCodecProvider``, a codec provider that includes
>   :ref:`Codecs <fundamentals-codecs>` that define how to

Reviewer: S: a little repetitive

```suggestion
- ``PojoCodecProvider``, which provides
  :ref:`Codecs <fundamentals-codecs>` that define how to
```

  encode and decode the data between the POJO format and BSON. The

>   provider also specifies which POJO classes or packages that the codecs

Author: changed all non-monospace to lowercase

  apply to.
- ``CodecRegistry`` instance that contains the codecs and other related information.
- ``MongoDatabase`` or ``MongoCollection`` instance configured to use the
  ``CodecRegistry``.
- ``MongoCollection`` instance that is created with the POJO document class
  bound to the ``TDocument`` generic type.

> The following steps perform each of the configuration requirements to
> use the sample POJO defined in the preceding section:

Reviewer: S: the wording here kinda implies that the step itself is the subject taking some action; you could reword so that the reader is the subject and the one taking action

```suggestion
Perform the following steps to meet the configuration requirements defined in the preceding section:
```

   ``CodecRegistry`` allows you to specify one or more codec providers to
   encode the POJO data. In this example, we call the following methods:

   - ``fromRegistries()`` to combine multiple ``CodecRegistry``
     instances into one instance
   - ``getDefaultCodecRegistry()`` to retrieve a ``CodecRegistry`` instance from a list of codec providers
   - ``fromProviders()`` to create a ``CodecRegistry`` instance from the ``PojoCodecProvider``

   .. _get-default-codec-registry-example:

   The following code shows how to instantiate the ``CodecRegistry``:

   .. code-block:: java

      // Include the following static imports before your class definition
      import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
      import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
      import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;

      ...

      CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));

#. Configure the ``MongoDatabase`` or ``MongoCollection`` instance to use the
   Codecs in the ``CodecRegistry``. You can configure either a database

>    or collection to specify the Codecs.

Reviewer: Q: same codec capitalization question as above

Summary
-------

> In this guide, we explain how to convert data between BSON and POJO fields

Reviewer: S: since the style guide suggests limiting "we", you could reword to something like:
```suggestion
This guide describes how to convert data between BSON and POJO fields
```

by performing the following tasks:

- Instantiate a ``PojoCodecProvider`` which contains codecs that define how to

>   encode/decode data between BSON and the POJO fields.

Reviewer: S: to remove the slash
```suggestion
  encode and decode data between BSON and the POJO fields.
```
